### PR TITLE
Remove the `strategy` package

### LINKED ISSUE #74: Remove the `strategy` package
We can only keep the `UnifyingStrategy` and `Aggregation` that should replace the `make_aggregation`, possibly put `matrixifying`, `reshape` and `unifying` bundled into one file `aggregation.py`.

### PR DESCRIPTION
**Pull request by ValerianRey**
- Fixes #74 

Description :
* Remove `ExtrapolatingStrategy` and its unit tests
* Remove `IsolatingStrategy` and its unit tests
* Remove `PartitioningStrategy` and its unit tests
* Rename `aggregation.py` to `aggregate.py`
* Move `UnifyingStrategy`, `Reshape` and `Matrixify` to `aggregate.py`
* Remove the `strategy` package in sources and unit tests
* Make `UnifyingStrategy`, `Reshape` and `Matrixify` protected
* Rename `_UnifyingStrategy` to `_AggregateMatrices`
* Replace function `make_aggregation` by transform `Aggregate`
* Remove property testers for strategies: now directly test `_AggregateMatrices` with functions
* Improve the randomness of jacobian matrix dicts generation in tests by using a random generator
* Fix mistake in `test_disunite_wrong_vector_length`
* Add docstrings to testing functions in `test_aggregate.py` when missing

TODO: should we add a test for `Aggregate`? We currently only test its components.

### CODE REVIEW

File: src/torchjd/autojac/_transform/aggregate.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    @property
+    def required_keys(self) -> set[Tensor]:
+        return self.transform.required_keys
+
+    @property
+    def output_keys(self) -> set[Tensor]:
+        return self.transform.output_keys
+
+
+class _UnifyingStrategy(Transform[JacobianMatrices, GradientVectors]):
```
Comment (PierreQuinton) [inline src/torchjd/autojac/_transform/aggregate.py:41]: This is now a weird name, I would basically make this `Aggregate` and the other one `_Aggregate`.

File: src/torchjd/autojac/backward.py [hunk 2]
```diff
@@ -81,7 +80,7 @@ def backward(
     jac = Jac(tensors, inputs, parallel_chunk_size, retain_graph)
 
     # Transform that aggregates the Jacobians.
-    aggregate = make_aggregation(UnifyingStrategy(A, inputs))
+    aggregate = Aggregate(A, inputs)
```
Comment (PierreQuinton) [inline src/torchjd/autojac/backward.py:83]: This is so much better, I love it.

File: src/torchjd/autojac/_transform/aggregate.py (hunk 1 above)
Comment (ValerianRey) [reply src/torchjd/autojac/_transform/aggregate.py]: I think `_AggregateMatrices` is better. It disambiguates with `Aggregate`.

**Review by PierreQuinton — APPROVED** [review]
LGTM, very nice!

**Pull request merged**